### PR TITLE
Fix/file upload deprecation 1490

### PR DESCRIPTION
**Pull request by kriswinbush**
## Description

Added deprecation demos for File Upload component
https://github.com/Teradata/covalent/issues/1490
### What's included?

- demo
- redirect to file input component

#### Test Steps

- [ ] `npm run serve`
- [ ] component -> file upload

#### General Tests for Every PR

- [ ] `npm run serve:prod` still works.
- [ ] `npm run tslint` passes.
- [ ] `npm run stylelint` passes.
- [ ] `npm test` passes and code coverage is not lower.
- [ ] `npm run build:lib` still works.

### CODE REVIEW

File: src/app/content/components/components.ts [hunk 1]
```diff
@@ -378,6 +378,23 @@ export const createComponentDetails: IComponentDetails[] = [
         (mod: any) => mod.TextEditorDemoModule,
       ),
   },
+  {
+    name: 'File Upload',
+    id: 'file-upload',
+    description: 'Upload files from your compuer or device',
```
Comment (aqupriyanka) [inline src/app/content/components/components.ts]: typo mistake - `computer`

File: src/app/content/components/component-demos/file-upload/demos/file-upload-demo-basic/file-upload-demo-basic.component.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<h2>Covalent</h2>
+<button mat-button routerLink="/components/file-input" routerLinkActive="active">Use Covalent file-input component</button>
+
+
+<hr>
+<h2>Native with Material</h2>
+<ng-container *ngIf="!file; else upload">
+  <button type="button" mat-raised-button (click)="fileInput.click()" [disabled]="disabled">
+    <mat-icon>attach_file</mat-icon>
+    'Choose a file...  <span [hidden]="!fileInput?.required">*</span>
```
Comment (aqupriyanka) [inline src/app/content/components/component-demos/file-upload/demos/file-upload-demo-basic/file-upload-demo-basic.component.html]: remove starting quote `'` before `Choose a file`

File: src/app/content/components/component-demos/file-upload/demos/file-upload-demo-basic/file-upload-demo-basic.component.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  </button>
+</ng-container>
+<ng-template #upload>
+  <button type="button" mat-raised-button (click)="uploadDoc()" [disabled]="disabled">
```
Comment (aqupriyanka) [inline src/app/content/components/component-demos/file-upload/demos/file-upload-demo-basic/file-upload-demo-basic.component.html]: we dont have `disabled` variable in component, can we either remove it or make it `false`

File: src/app/content/components/component-demos/file-upload/file-upload.component.html [hunk 4]
```diff
@@ -0,0 +1 @@
+<p>file-upload works!</p>
```
[the remark this answers is not in the document]
Comment (kriswinbush) [reply src/app/content/components/component-demos/file-upload/file-upload.component.html]: done

File: src/app/content/components/components.ts (hunk 1 above)
Comment (kriswinbush) [reply src/app/content/components/components.ts]: done

File: src/app/content/components/component-demos/file-upload/demos/file-upload-demo-basic/file-upload-demo-basic.component.html (hunk 2 above)
Comment (kriswinbush) [reply src/app/content/components/component-demos/file-upload/demos/file-upload-demo-basic/file-upload-demo-basic.component.html]: removed example

File: src/app/content/components/component-demos/file-upload/demos/file-upload-demo-basic/file-upload-demo-basic.component.html (hunk 3 above)
Comment (kriswinbush) [reply src/app/content/components/component-demos/file-upload/demos/file-upload-demo-basic/file-upload-demo-basic.component.html]: removed example

**Pull request merged**